Sync v2-7-stable with v2-7-test to release 2.7.1

ephraimbuddy opened the pull request:
Time for `2.7.1rc1`!

Note that some provider changes where cherrypicked to solve some conflicts in those simplify code PR

pankajkoti committed:
Validate SqoopHook connection string and disable extra options from public hook methods (#33039)

* Validate SqoopHook connection string and disable extra options from hook methods

Check that the connection string constructed using the connection's
`host`, `port` and `schema` does not contain query params as it is not
intended. Additionally, also disable the `extra_import_options` and
`extra_export_options` arguments accepted directly by the hook
methods but accept it as a param via the hook constructor when
initialising the hook or by passing it in hook_params when initialising
the hook from operators.

* Propogate missing hook param changes to the operator

* Remove test for invalid port with query param as ports are integers and not accepted by databases

* Add 4.0.0 to provider.yaml

(cherry picked from commit 59f5f58e2bfcc8c12b0b44d6c6ce5cfed60b8867)

ahidalgob committed:
Fix test_example_dags (#32714)

By going up to `parents[3]` we were going outside the repository root,
luckily(or unluckily the repo folder is also named `airflow` so the
pattern `airflow/**/example_dags/example_*.py` still worked,
but `tests/system/providers/**/example_*.py` wasn't being used.

This discovered 2 new errors:

- `example_local_to_wasb.py` was trivial to fix

- `example_redis_publish.py`is more interesting: this one fails because
`RedisPubSubSensor` constructor calls Redis.pubsub().subscribe(), which
just hangs and DagBag fails with timeout. For now I'm just deleting this
operator from the example.

(cherry picked from commit c048bd5d9e98b05a8c2b68af0ef1853a700888f7)

Owen-CH-Leung committed:
Upgrade Elasticsearch to 8 (#33135)

(cherry picked from commit ad9d8d46b6ee3a7d8e6665d2a6f5c6660063f281)

eumiro committed:
Refactor: Simplify code in providers/cncf (#33230)

(cherry picked from commit 056f1670fe85a6f9d70cd72ec2e88287e20e6827)

ephraimbuddy committed:
Fix pydantic warning about `orm_mode` rename (#33220)

* Fix pydantic warning about `orm_mode` rename

Pydantic 2 renamed orm_mode to from_attributes. This was missed during the upgrade to pydantic 2 and it gives excessive warning about the rename.
This PR fixes it

* Also rename from_orm to model_validate and use model_dump instead of dict

* Fix Pydantic 1.x compatibility

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 75bb04bb6f37dab4083190e4a3d6436ace1cac31)

eumiro committed:
Refactor: Simplify code in providers/amazon (#33222)

(cherry picked from commit 83bd60fd97d4ca622adcbd7898d88880fee43054)

eumiro committed:
Refactor: Simplify dict manipulation in metrics (#33264)

(cherry picked from commit 01a6c1e7f8b0ed736d7c6f661e0c29ec023016b2)

eumiro committed:
Refactor: Simplify code in www (#33270)

(cherry picked from commit 369b9bc947215d396570b06c3a5e7b982f234e76)

eumiro committed:
Refactor: Simplify code in serialization (#33266)

(cherry picked from commit 63c5df89d9a6db7c8f3c8a0a252588b98fb0b1a1)

eumiro committed:
Refactor: Simplify code in utils (#33268)

(cherry picked from commit cc8519d1bedd665d89834343171ecf62e72d8f2a)

eumiro committed:
Refactor: Simplofy code in dev (#33294)

(cherry picked from commit 3b313e218d17ba07757ce08acdceab7620fb4fd6)

ferruzzi committed:
D205 Support - Root files (#33297)

* D205 Support - Root files

Updates setup.py, airflow/configuration.py, and airflow/exceptions.py

* missed one

(cherry picked from commit 4755fe40350c83fafef4525bf47040e1a1f0f07a)

pankajkoti committed:
Remove the replace to <br> tag in Provider's view (#33326)

(cherry picked from commit 23d542462a1aaa5afcd36dedc3c2a12c840e1d2c)

ferruzzi committed:
D401 Support - Airflow/api thru Airflow/auth (#33333)

(cherry picked from commit b657ae914e3783f28be16e7d3acdc86cb3d37e83)

ferruzzi committed:
D401 Support - airflow/callbacks thru airflow/decorators (#33335)

(cherry picked from commit 396fd3cae91df2d16eeb9d399784687e87e47035)

ferruzzi committed:
D401 Support - airflow/example_dags thru airflow/listeners (#33336)

(cherry picked from commit d0c94d6bee2a9494e44f29c2c242c956877e9619)

ferruzzi committed:
Fix some missing type hints (#33334)

(cherry picked from commit 3c48dc714b65d2ed5e29c8b29fc58cc4a3afabf8)

ferruzzi committed:
D401 Support - Utils (#33339)

(cherry picked from commit 827962878e6fb39e014639d83cff7d0881595ecb)

eumiro committed:
Refactor: Simplify code in tests (#33293)

(cherry picked from commit d2c0bbebacd1c58d357d647707a1bdb0c5b3b425)

ferruzzi committed:
D401 Support - Macros to Operators (Inclusive) (#33337)

* D401 Support - airflow/macros thru airflow/operators

* fix static checks

(cherry picked from commit 2efb3a64f5446a5103c4107e4d2db4c629344faa)

ferruzzi committed:
D205 Support - WWW (#33298)

* D205 Support - WWW

* fix broken link

(cherry picked from commit 64c2a56faf31d5f7e38e7b55e9eb4c38ced54c59)

ferruzzi committed:
D401 Support - Secrets to Triggers (Inclusive) (#33338)

(cherry picked from commit 44a752aa96d5aa4263f4ac1725642084fd612a48)

eumiro committed:
Refactor: Simplify code in scripts (#33295)

(cherry picked from commit 50a6385c7fed0fd5457886d0ffdd5040f6a8d511)

eumiro committed:
Refactor: Simplify code in settings (#33267)

(cherry picked from commit 95e9d83720584e2c098242dd60aa54f388dae1de)

ferruzzi committed:
D401 Support - Root Files (#33352)

(cherry picked from commit 5e1e5fa637aa90b92e4b0e1d62c577461ea92369)

eumiro committed:
Refactor: Simplify code in configuration.py (#33160)

(cherry picked from commit 21656fcfe84b2e749b6b7bc9a3da4826c1af522f)

Lee-W committed:
respect soft_fail argument when ExternalTaskSensor runs in deferrable mode (#33196)

(cherry picked from commit a1b5bdb25a6f9565ac5934a9a458e9b079ccf3ae)

ohaibbq committed:
[Models] [Postgres] Check if the dynamically-added index is in the table schema before adding (#32731)

* Check if the index is in the table schema before adding

* add pre-condition assertion

* static checks

* Update test_models.py

* integrate upstream auth manager changes

(cherry picked from commit 2950fd768541fc902d8f7218e4243e8d83414c51)

pierrejeambrun committed:
Fix DagFileProcessor interfering with dags outside its ``processor_subdir`` (#33357)

* Fix standalone DagProcessor interfering with DAG outsite of its subdir

* Add tests

* Update code review

(cherry picked from commit 35b18306a4928152fd1834964fc8ce0033811817)

AndreasAlbertQC committed:
Fix config description for base_log_folder (#33388)

(cherry picked from commit 08565dcb8b296609a22955e79291ae9dfb51b5bb)

ferruzzi committed:
D205 Support - Stragglers (#33301)

(cherry picked from commit 9bf68adfa0d929c4400d3e9f02cd2feaa4932f59)

mobuchowski committed:
openlineage: do not try to redact Proxy objects from deprecated config (#33393)

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>
(cherry picked from commit 8e738cd0ad0e7dce644f66bb749a7b46770badee)

ephraimbuddy committed:
Fix getting correct commit from multiple referenced PR (#33411)

When a PR is referenced by other PRs, our dev tool for getting the correct
commit lists the latest commit when looking for the commmit sha but we should get the oldest.

(cherry picked from commit 5b104a9020510c75f1d5a7a211fd0c7cb1b45070)

potiuk committed:
More complete fix for dev release scripts to filter commit for PR (#33418)

This is a more complete fix to #33411. This is also a follow up on
earlier implementation of #33261 that addressed checking if PRs
are merged. This one applies the same pattern to finding commit
but also improves it by checking if the (#NNNNNN) ends the subject
- so even if the PR is in the same form in the message, it will be
filtered out.

The previous "--reverse" quick fix in #33411 had potential of problem in
case there were releated PRs merged before the original PR (which is
quite posssible when you have a series of PRs referring to each other.

(cherry picked from commit 3766ab07d7061e946329443ed9a073ad3b41f457)

jscheffl committed:
Fix handling of default value and serialization of Param class (#33141)

(cherry picked from commit 489ca1494691f6a0b717f9312e0ab8a2a7f76d96)

eumiro committed:
Refactor str.startswith with tuples (#33292)

(cherry picked from commit d747a7924bca4b1e44003fe0932c28d7b504b088)

eumiro committed:
Import utc from datetime and normalize its import (#33450)

(cherry picked from commit bfe08a79db8130c499883f014121be570ec071bd)

pierrejeambrun committed:
Fix Cluster Activity Health margin (#33456)

(cherry picked from commit 2ae6d4bf5d5dab070d806a22071b23b4f40946f3)

ferruzzi committed:
Static check fixes (#33462)

(cherry picked from commit 5ee1bcbef992102b3957db39a90254dfc9c4adeb)

pankajkoti committed:
Use absolute path in README.md for broken links on pypi listing (#33473)

Observed a couple of links are broken on our pypi project listing
https://pypi.org/project/apache-airflow/. They work fine when
viewed on GitHub but those relative links don't work fine on the
pypi index. Also, applied a few quick-fixes that my PyCharm
suggested for the README.md file

(cherry picked from commit 5a37025eb0a2acc90c41a1f9701a70521f361e71)

eumiro committed:
Consolidate import and usage of pandas (#33480)

(cherry picked from commit 8e88eb8fa7e1fc12918dcbfcfc8ed28381008d33)

Aakcht committed:
Fix secrets backend docs (#33471)

(cherry picked from commit 996d8c5c930e6e7f800c0c7a5c0b1ade7516c89f)

uranusjr committed:
Allow timetable to slightly miss catchup cutoff (#33404)

(cherry picked from commit a6299d47eac9cba23d54e5b1f3ff589e76871eae)

hussein-awala committed:
Fix a bug in formatDuration method (#33486)

(cherry picked from commit eed09daad3278706da29c8f172899e67ecadc40a)

potiuk committed:
Remove openlineage exclusion (#33491)

Now that Airflow 2.7.0 is released, we can remove exclusion that
we had for openlineage which prevented from using it as
dependency of Airflow in CI.

(cherry picked from commit 008f2335f8eb40624e164a2cc608148bd356d17e)

pierrejeambrun committed:
Do not return ongoin dagrun when a end_date is less than utcnow (#33488)

(cherry picked from commit 7c51c87503004d57fe726dae2ab830a5ffd9b64b)

vijayasarathib committed:
Fix broken link in Modules Management page (#33499)

fixing broken link in Modules Management

(cherry picked from commit 02d3378fc937bacabed307b5b9509b729683bf4a)

hussein-awala committed:
Fix rendering the mapped parameters when using `expand_kwargs` method (#32272)

* Fix rendering the mapped parameters in the mapped operator

Signed-off-by: Hussein Awala <hussein@awala.fr>

* add template_in_template arg to expand method to tell Airflow whether to resolve the xcom data or not

* fix dag serialization tests

* Revert "fix dag serialization tests"

This reverts commit 191351cda7b51bc6d49e7fcee5ab8ccd6cd219f5.

* Revert "add template_in_template arg to expand method to tell Airflow whether to resolve the xcom data or not"

This reverts commit 14bd392c7e5c6f25ce25d61dfd440e9b27c1bc2e.

* Fix ListOfDictsExpandInput resolve method

* remove _iter_parse_time_resolved_kwargs method

* remove unnecessary step

---------

Signed-off-by: Hussein Awala <hussein@awala.fr>
(cherry picked from commit d1e6a5c48d03322dda090113134f745d1f9c34d4)

Taragolis committed:
Use `dialect.name` in custom SA types (#33503)

* Use `dialect.name` in custom SA types

* Fix removed import

(cherry picked from commit 46aa4294e453d800ef6d327addf72a004be3765f)

kaxil committed:
Fix typo in `db upgrade` warning message (#33510)

`db updgrade` to `db upgrade`

(cherry picked from commit bfba17bb37feea74247a06aaa65560d125247fee)

ephraimbuddy committed:
Update version_added field for configs in config file (#33509)

The otel was 2.6.0 as can be seen here: https://github.com/apache/airflow/pull/30160 and
the internal api and related were 2.6.0 even though they were under a flag. These need to be updated
except we want to modify the check tool to skip these ones

(cherry picked from commit ba4f541fba656acb6c35469c97a1e1bfc79f2d96)

potiuk committed:
Remove Pydantic limitation for version < 2 (#33507)

We already fixed all deprecation warnings for Pydantic 2 and we can
thus remove Pydantic 2 limitation. Even if we are waiting for other
dependencies (aws-sam-translator) it should be save to remove the
limit - we will get Pydantic 2 when aws-sam-translate new version
is released in a week or two (Pydantic 2 support has been added
last week in
https://github.com/aws/serverless-application-model/pull/3282)

(cherry picked from commit 754a4ab396e6e6002f79a716f93ed71712393687)

potiuk committed:
Fix typo in release notes (#33521)

(cherry picked from commit e69aae1304b478b76d9088729ce6d4df5448e5c0)

jedcunningham committed:
Move license templates out of repo root (#33515)

(cherry picked from commit f89af2163fdc907aac8ebb9d1d0ddadf6e5c4806)

ldacey committed:
Fix OpenLineage link in New Features notes (#33513)

The label is provider:openlineage instead of API-53

(cherry picked from commit de17b939eb37d0bb9e75ec48be2a0e7159e4cd1c)

hussein-awala committed:
Stop adding values to rendered templates UI when there is no dagrun (#33516)

(cherry picked from commit d9814eb3a2fc1dbbb885a0a2c1b7a23ce1cfa148)

eumiro committed:
Simplify code around enumerate (#33476)

(cherry picked from commit bcefe6109bcabd9bd6daf8b44f7352adda1ed53d)

eumiro committed:
Replace OrderedDict with plain dict (#33508)

(cherry picked from commit 63e6eab785f8313be6b5e28056215947bb8ed8a3)

eumiro committed:
Simplify 'X for X in Y' to 'Y' where applicable (#33453)

(cherry picked from commit 7700fb12cc6c7a97901662e6ac6aa1e4e932d969)

hussein-awala committed:
Set strict to True when parsing dates in webserver views (#33512)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 4390524a41fdfd2d57f1d2dc98ad7b4009c8399e)

eumiro committed:
Refactor: Simplify code in smaller providers (#33234)

(cherry picked from commit a91ee7ac2fe29f460a4e4b0d8c1346f40672be43)

eumiro committed:
Replace repr() with proper formatting (#33520)

(cherry picked from commit abef61ff3d6b9ae8dcb7f9dbbea78a9648a0c50b)

eumiro committed:
Simplify conditions on len() in other providers (#33569)

(cherry picked from commit 1cdd82391e0f7a24ab7f0badbe8f44a54f51d757)

eumiro committed:
Consolidate import and usage of itertools (#33479)

(cherry picked from commit 95a930bc0a720c5548e4fa2e1f74e25f12e9ae1d)

eumiro committed:
Simplify conditions on len() in jobs (#33568)

(cherry picked from commit eb56473b40d2717758613ead1cdf712c33347082)

eumiro committed:
Simplify conditions on `len()` in utils (#33567)

(cherry picked from commit a1e6cd411af9a922a35d770fbe576d8e892f44e1)

hussein-awala committed:
Create a new method used to resume the task in order to implement specific logic for operators (#33424)

* Create a generic method used to resume the task in order to implement specific logic for each operator

* Provide resume_execution args

* add a unit test

* Fix soft fail after a trigger timeout

* wrap the exception with a str to get the message

* Apply suggestions from code review

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 852f85bc61c04fcf39618b8a7ee5ec885f8fde28)

potiuk committed:
Add MySQL 8.1 to supported versions. (#33576)

* Add MySQL 8.1 to supported versions.

Anticipating Lazy Consensus to be reached we add 8.1 version
of MySQL to supported versions.

* Apply suggestions from code review

(cherry picked from commit 825f65f67e0732c84a1978c342a4f77d152636b9)

potiuk committed:
Fix dependencies for celery and opentelemetry for Python 3.8 (#33579)

We used to have problems with `pip` backtracking when we relaxed
too much open-telemetry dependencies. It turned out that the
backtracting was only happening on Python 3.8 and that it was
ultimately caused by conflict between importlib_metadata between
Airflow and newer versions of opentelemetry (we had <5 for Python
3.8, they had >6 for all versions. The reason for limiting it in
Airflow was Celery that was not working well with importlib 5.

Since Celery 5.3 solved the problems (released 6th of June) we can
now relax the importlib_metadata limit and set Celery to version >=
5.3.0) which nicely resolves the conflict and there is no more
backtracking when trying to install newer versions of opentelemetry
for Python 3.8.

Fixes: #33577
(cherry picked from commit ae25a52ae342c9e0bc3afdb21d613447c3687f6c)

vandonr-amz committed:
make `conf.set` case insensitive (#33452)

* make `conf.set` case insensitive

`conf.get` is insensitive (it converts section and key to lower case)
but set is not, which can lead to surprising behavior
(see the test, which is not passing without the fix).
I suggest that we override set as well to fix that.
Any value that was set before with upper case was unreacheable.

* fix remove_option as well

* away with the str()

* add significant change newsfragment

(cherry picked from commit abbd5677bab4a84b1d35e7723c7dfbb155ca9144)

potiuk committed:
Improve detection of when breeze CI image needs rebuilding (#33603)

* Improve detection of when breeze CI image needs rebuilding

Previously we have been using provider.yaml file modification as
a sign that the docker image needs rebuilding when starting image.
However just modification of provider.yaml file is not a sign
that the image needs rebuilding. The image needs rebuilding when
provider dependencies changed, but there are many more reasons why
provider.yaml file changed - especially recently provider.yaml
file contains much more information and dependencies are only part
of it. Provider.yaml files can also be modified by release manager
wnen documentation is prepared, but none of the documentation
change is a reason for rebuilding the image.

This PR optimize the check for image building introducing two
step process:

* first we check if provider.yaml files changed
* if they did, we regenerate provider dependencies by manully
  running the pre-commit script
* then provider_dependencies.json is used instead of all providers
  to determine if the image needs rebuilding

This has several nice side effects:

* the list of files that have been modified displayed to the
  user is potentially much smaller (no provider.yaml files)
* provider_dependencies.json is regenereated automatically when
  you run any breeze command, which means that you do not have
  to have pre-commit installed to regenerate it
* the notification "image needs rebuilding" will be printed less
  frequently to the user - only when it is really needed
* preparing provider documentation in CI will not trigger
  image rebuilding (which might occasionally fail in such case
  especially when we bring back a provider from long suspension
  like it happened in #33574

* Update dev/breeze/src/airflow_breeze/commands/developer_commands.py

(cherry picked from commit ac0d5b3dbe731605af38018ce7ce970ffded539a)

vincbeck committed:
Add a fallback in case no first name and last name are set (#33617)

(cherry picked from commit 62b917a6ac61fd6882c377e3b04f72d908f52a58)

hussein-awala committed:
Revert "respect soft_fail argument when ExternalTaskSensor runs in deferrable mode (#33196)" (#33458)

This reverts commit a1b5bdb25a6f9565ac5934a9a458e9b079ccf3ae.

(cherry picked from commit 85a567790456732747ddccd37f690d7f3a56d2ba)

ephraimbuddy committed:
Automatically update the REVISIONS_HEADS_MAP in db.py (#33616)

* Automatically update the REVISIONS_HEADS_MAP in db.py

This has been the responsibility of the release manager and manually done.
In 2.7.0, I mistakenly added a wrong value for the revision which means that
anyone upgrading with `airflow db migrate --version 2.7.0` will end up having incorrect
migration.
This PR automates this and always updates it to the correct value through pre-commit

* fixup! Automatically update the REVISIONS_HEADS_MAP in db.py

* Account for missing versions in revision heads map

* Provide the full path of the DB file in message

* Make REVISION_HEADS_MAP private

* Only add from 2.0.0 up

(cherry picked from commit 513c1d2c650c800978f37ab253e230c82777bbd8)

eumiro committed:
Fix typos (double words and it's/its) (#33623)

(cherry picked from commit a54c2424df51bf1acec420f4792a237dabcfa12b)

ephraimbuddy committed:
Update the release date for 2.7.0

SamWheating committed:
Bind engine before attempting to drop archive tables (#33622)

(cherry picked from commit 911cf466218bcd548519a50c9a32c9df58ec8b2e)

potiuk committed:
Upgrade botocore/aiobotocore minimum requirements (#33649)

Botocore has a very peculiar process of releasing new version
every day, which means that it gives `pip` hard time to figure
what will be the non-conflicting set of packages when we have
too low of a minium version set as requirement.

Since we had > 1.24 that means that `pip` had to consider
more than 340 versions for botocore, but also for related
mypy packages and also a number of aiobotocore packages when
resolving eager-upgrade.

We limit all the relevant packages to 1.28 as minimum version
now, and we should continue doing that regularly in the future.

(cherry picked from commit 5f504e9a17353259e70bd3ed54f8edd2e465882c)

LipuFei committed:
Fix 2.7.0 db migration job errors (#33652)

* Fix circular import from kubernetes_executor

* Do not validate SQL when initialising Sentry

---------

Co-authored-by: Lipu Fei <lipu.fei@kpn.com>
(cherry picked from commit 4bdf908041159c1d6ed019469190e08297928393)

vincbeck committed:
Add back `get_url_for_login` in security manager (#33660)

* Add back `get_url_for_login` in security manager

* Rename FAB_SECURITY_MANAGER_CLASS to SECURITY_MANAGER_CLASS

(cherry picked from commit b1cdab39f85879f2b0189685c0b7f7dcdc8d62f0)

Taragolis committed:
Sort data before groupby in TIS duration calculation (#33535)

(cherry picked from commit 79b8cfc0fa77f11491fc1de4d5f009e176aa7c3a)

potiuk committed:
Bring back hive support for Python 3.11 (#32607)

Update airflow/providers/apache/hive/CHANGELOG.rst

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 08188f80f4bb6c225eaf4f58e2742c982a230652)

ivan-afonichkin committed:
Set context inside templates (#33645)

* Set context inside templates

---------

Co-authored-by: Ivan Afonichkin <ivanafonichkin@Ivans-MacBook-Pro.local>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 9fa782f622ad9f6e568f0efcadf93595f67b8a20)

Michalosu committed:
SECURITY_MANAGER_CLASS should be a referrence to class, not a string (#33690)

(cherry picked from commit f971ba2f2f9703d0e1954e52aaded52a83c2f844)

SamWheating committed:
Treat dag-defined access_control as authoritative if defined (#33632)

(cherry picked from commit 370348a396b5ddfe670e78ad3ab87d01f6d0107f)

Taragolis committed:
Add limit 1 if required first value from query result (#33672)

(cherry picked from commit e8ba57968d239cf4c399cdc81c0857790c645c6c)

Bisk1 committed:
Fix cleaning zombie RESTARTING tasks (#33706)

* Fix cleaning zombie RESTARTING tasks

* Fix test

* Improve naming

Renamed to adoptable_states because adoption is preferrable choice and reset is fallback.
Also "resettable" might be confused with RESTARTING itself.

---------

Co-authored-by: daniel.dylag <danieldylag1990@gmail.com>
(cherry picked from commit 5c35786ca29aa53ec08232502fc8a16fb1ef847a)

Lee-W committed:
Respect "soft_fail" argument when "poke" is called (#33401)

* feat(sensors/base): raise AirflowSkipException if soft_fail is set to True and exception occurs after running poke()

* test(sensor/base): add test case for respecting soft_fail option when other kinds of exception is raised

(cherry picked from commit d91c481ac9051316d0e61b02eda5a7e21bb6ac5b)

SamWheating committed:
Add documentation explaining template_ext (and how to override it) (#33735)

* Add documenation explaining template_ext (and how to override it)

* fix static checks

(cherry picked from commit a5aa16184d957ac0e4bfa4fdef15b32ffd312079)

hussein-awala committed:
Use f-string instead of  in Airflow core (#33753)

(cherry picked from commit 272b40aaad33116bd0b0fdcad15187493d9a44e7)

potiuk committed:
Limit hive provider check for Python 3.11 temporarily (#33774)

In order to generate constraints, we need to temporarily limit
also hive provider. There is a gap between wnen we added it
in airflow setup and when we can generate constraints for
the released providers from PyPI - we need to release the provider
similarly like we have to do it for yandex.

Therefore - until the upcoming hive provider is released (in 3 days)
we need to limit hive from being considered in Python 3.11 consstraint
generation for providers from PyPI

(cherry picked from commit 984ba22e6e2a8c70b780d11c0bbaeb1dc0d6f14d)

Lee-W committed:
Respect "soft_fail" for core async sensors (#33403)

* fix(sensors): ensure that DateTimeSensorAsync, TimeDeltaSensorAsync, TimeSensorAsync respect soft_fail

* refactor(sensors): move the soft_fail checking logic from DateTimeSensorAsync, TimeDeltaSensorAsync, TimeSensorAsync to DateTimeTrigger

* test(triggers/temporal): add test case for DateTimeSensorAsync respects soft_fail

* fix(triggers/temporal): use the original error message with skipping postfix as message for AirflowSkipException

* Revert "fix(triggers/temporal): use the original error message with skipping postfix as message for AirflowSkipException"

This reverts commit a6d803303bf71a84e9e59e94d9c088e3120bedb5.

* Revert "test(triggers/temporal): add test case for DateTimeSensorAsync respects soft_fail"

This reverts commit 50e39e08a415685ace788ae728397a199c21e82b.

* Revert "refactor(sensors): move the soft_fail checking logic from DateTimeSensorAsync, TimeDeltaSensorAsync, TimeSensorAsync to DateTimeTrigger"

This reverts commit 985981a269cea68da719d6fd1c60bedd9a7e5225.

* Revert "fix(sensors): ensure that DateTimeSensorAsync, TimeDeltaSensorAsync, TimeSensorAsync respect soft_fail"

This reverts commit b2f2662ae1a11ea928aad57acd2892c763c2db25.

* fix(sensors): move core async sensor trigger initialization to __init__ if possible

(cherry picked from commit 9ce76e321f9792690a8d93d5ecb4df9bdaf8fac9)

potiuk committed:
Limit Redis dependencies to match celery limits (#33773)

Redis 5 relased last week breaks celery, celery is limiting it for
now and will resolve it later, we should similarly limit redis on
our side to limit redis for users who will not upgrade to celery
that will be released shortly.

Fixes: #33744
(cherry picked from commit 3ba994d8f4c4b5ce3828bebcff28bbfc25170004)

jedcunningham committed:
Fix version heads map pre_commit rule (#33749)

Only run it when the map may have changed (migration or core version
change), and also only check migration files (ignore things like
pycache).

(cherry picked from commit a746defd6ba123b74d87ace4de01fd0a25932a64)

potiuk committed:
Remove "eager upgrade" from PROD image completely (#33784)

There were still some left-overs of EAGER_UPGRADE in PROD image
building. Howwever "eager upgrade" only makes sense for CI images.
PROD images when being built should use eager upgrades as they
are produced in the CI image step.

This PR does the following:

* removes eager upgrade parameters from PROD image
* instead, prod image build has a new flag for installing
  the images: --use-constraints-for-context-packages which will
  automatically use constraints from "docker-context-files" if
  they are present there.
* modifies the CI workflows to upload constraints as artifacts
  and download them for PROD image build when "eager upgrade"
  has been used and directs it to use "source" constraints
* adds back support to "upgrade to newer dependencies" label
  that makes it easy to test "eager upgrade"

As the result, when PROD image is build in CI:

* when regular PR is run, it will use latest github "source" constraints
* whwn "eager upgrade" PR is run, it will use the eager-upgrade
  constraints that were generated during CI build

(cherry picked from commit 2b1a1944aa5cc84c94df523309eea30890d4fb74)

potiuk committed:
Bump airflow version to 2.7.1

potiuk committed:
Use PyPI constraints for PROD image in non-main branch (#33789)

When we are building PROD image in CI for non main branch, we are
installing providers from PyPI rather than building them locally
from sources. Therefore we should use `PyPI` constraints for
such builds not the "source" constraints (they might differ).

This PR adds two steps:

* In the CI build, when we do not build providers we generate
  PyPI constraints additionally to source constraints
* In the PROD build we use the PyPI constraints in case we
  do not build providers locally

(cherry picked from commit f9276f0597d5ad91c98d0db36d16d799ee2d4f21)

boushphong committed:
Fix UI DAG counts including deleted DAGs (#33778)

(cherry picked from commit 64948fa7824d004e65089c2d159c5e6074727826)

ephraimbuddy committed:
Update version to 2.7.1

ephraimbuddy committed:
Update RELEASE_NOTES.rst

eladkal committed:
Replace AIP-42 label with area:dynamic-task-mapping label (#33870)

(cherry picked from commit 6aeda2ce2b267b36b90cc1eea5cc6bcf5cf2085b)

potiuk committed:
Suspend qubole provider (#33889)

* Suspend qubole provider

Qubole has been acquired and seems that maintainers have left the project
https://github.com/qubole/qds-sdk-py#where-are-the-maintainers-
the package has been unmaintained for a long time and it's likely no-one uses it
until someone steps up to maintain it, we suspend it

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit 1f0e673ce3db5925dc1075e921ae204aa653118a)

ephraimbuddy committed:
Fix MappedTaskGroup tasks not respecting upstream dependency (#33732)

* Fix MappedTaskGroup tasks not respecting upstream dependency

When a MappedTaskGroup has upstream dependencies, the tasks in the group don't wait for the upstream tasks
before they start running, this causes the tasks to fail.
From my investigation, the tasks inside the MappedTaskGroup don't have upstream tasks while the
MappedTaskGroup has the upstream tasks properly set. Due to this, the task's dependencies are met even though the Group has
upstreams that haven't finished.
The Fix was to set upstreams after creating the task group with the factory
Closes: https://github.com/apache/airflow/issues/33446

* set the relationship in __exit__

(cherry picked from commit fe27031382e2034b59a23db1c6b9bdbfef259137)

potiuk committed:
Add minimum requirement for sqlalchemy to 1.4.24 (#33892)

Some of the recent sqlalchemy 2 changes used features tha were
added in 1.4.24 sqlalchemy (session.scalar).

We need to bump the minimum version to avoid accidental problems
with people upgrading and not bumping sqlalchemy nor using
constraints

Fixes: #33887
(cherry picked from commit bfab7daffedd189a85214165cfc34944e2bf11c1)

abhishekbhakat committed:
Raise 404 from Variable PATCH API if variable is not found (#33885)

* Raise variable not found if session returns empty

* Added detail to the exception for json reponse

* tests for patch api when variable doesn't exist

* Dropped fstring

* Unify varialbe not found message

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 701c3b80107adb9f4c697f04331c1c7c4e315cd8)

potiuk committed:
Clarify that DAG authors can also run code in DAG File Processor (#33920)

* Clarify that DAG authors can also run code in DAG File Processor

Small addition to our security model - it was not entirely clear
that DAG authors can also execute code in DAG File Processor and
that DAG File Processor can be run in standalone mode effectively
physically separating machines where scheduler is run and where
the code modified by DAG authors gets parsed.

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>
(cherry picked from commit 1dc6ba05581dba349f0b81163263ad6b0fa18b06)

Lee-W committed:
fix(sensors): move trigger initialization from __init___ to execute (#33926)

in https://github.com/apache/airflow/pull/33403, we move trigger initialization to __init__
which causes a failure for one uses template variable

(cherry picked from commit eaa6126e45ab9869c11c59363e52490aa6dff91f)

potiuk committed:
Reorganize devel_only extra in airflow's setup.py (#33907)

The `devel_only` extra puts together all dependencies that are
needed for CI image and in order to run tests in local virtualenv
for various test cases of ours - but they are not needed as dependencies
of particular providers. They were a little "bag of everything"
and they were hiding some unused dependencies or dependencies that
were either unused or they were actually provider dependencies already.

For example we had qds-sdk dependency there which was really the
qubole provider dependency and it held us back from removing
deprecated boto library from CI image (removed in #33889).

This PR organizes the dependency a bit better:

* split it to logical groups
* removes some unused dependencies
* moves "amazon" mypy dependency from providers to here

At later stage we will move the provider ones into "[devel]" extras
of the providers as part of provider decooupling, but this
will require a bit more changes in CI image building and some
documentation update for developers. This is an intermediate step
to organize it better.

(cherry picked from commit b497234eda4b4bdfda0b09e3674012c6ebebaf2b)

SamWheating committed:
Bumping FAB to 4.3.4 in order to fix issues with filters (#33931)

(cherry picked from commit ba261923d4de90d0609344843554bf7dfdab11c6)

SamWheating committed:
Differentiate 0 and unset as a default param values (#33965)

(cherry picked from commit c51901afc14bf82ff678226a7b8e32a972eff924)

potiuk committed:
Upgrade to Pydantic 2 (#33956)

The only blocking factor to migrate to Pydantic 2 was the
aws-sam-translator which was transitive dependency to
`moto[cloudformation]` via `cfn-lint` and we do not really need
everything in that extra - used only for testing.

While aws-sam-translator is already preparing to release Pydantic 2
compatible version, we do not want to wait - instead we replace the
cloudformation extra with openapi_spec_validator and jsonschema
needed by the cloudformation tests.

(cherry picked from commit 1cda0c34b9ab094bd17249bed053b1fa1c52e026)

potiuk committed:
Reinstall CI depdendencies from scratch (#33978)

With latest change enabling Pydantic #33956 some old dependencies
(aws-sam-translator) remained in the CI image from cached
installation and they are breaking `pip check` when refreshing
the image cache. This PR bumps EPOCH numbers so that the dependencies
are not installed from cache and we have a clean, new image with
just those depencies we need.

(cherry picked from commit dd7cc871605ad1239b48516c3df889cc7e3f19a8)

pankajkoti committed:
Use a trimmed version of README.md for PyPI (#33637)

(cherry picked from commit 88a7a70549a9b35e751fa4cf212714c61a237bf5)

potiuk committed:
Explain the users how they can check if python code is top-level (#34006)

Many users have problem with it. Adding the way how they can
check it easily.

(cherry picked from commit 9702a14dd59fb842eec720594f470b10b57b7a09)

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,112 @@
+- Fix ``MappedTaskGroup`` tasks not respecting upstream dependency (#33732)
+- Add limit 1 if required first value from query result (#33672)
+- Fix UI DAG counts including deleted DAGs (#33778)
+- Add documentation explaining template_ext (and how to override it) (#33735)
```
This is docs only.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,112 @@
+- Use ``dialect.name`` in custom SA types (#33503)
+- Do not return ongoing dagrun when a ``end_date`` is less than ``utcnow`` (#33488)
+- Fix a bug in ``formatDuration`` method (#33486)
+- make ``conf.set`` case insensitive (#33452)
```
```suggestion
- Make ``conf.set`` case insensitive (#33452)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,112 @@
+- Fix a bug in ``formatDuration`` method (#33486)
+- make ``conf.set`` case insensitive (#33452)
+- Allow timetable to slightly miss catchup cutoff (#33404)
+- Respect "soft_fail" argument when "poke" is called (#33401)
```
```suggestion
- Respect ``soft_fail`` argument when ``poke`` is called (#33401)
```

pankajkoti committed:
Remove print introduced in PR #32261 (#34008)

(cherry picked from commit 2232361c7cfd15f8d5e4384c06281cbd10071f42)

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,112 @@
+- Respect "soft_fail" argument when "poke" is called (#33401)
+- Create a new method used to resume the task in order to implement specific logic for operators (#33424)
+- Fix DagFileProcessor interfering with dags outside its ``processor_subdir`` (#33357)
+- Remove the replace to <br> tag in Provider's view (#33326)
```
```suggestion
- Remove the unnecessary ``<br> `` text in Provider's view (#33326)
```

Or something.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,112 @@
+- Create a new method used to resume the task in order to implement specific logic for operators (#33424)
+- Fix DagFileProcessor interfering with dags outside its ``processor_subdir`` (#33357)
+- Remove the replace to <br> tag in Provider's view (#33326)
+- respect soft_fail argument when ExternalTaskSensor runs in deferrable mode (#33196)
```
```suggestion
- Respect ``soft_fail`` argument when ExternalTaskSensor runs in deferrable mode (#33196)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,112 @@
+- Remove the replace to <br> tag in Provider's view (#33326)
+- respect soft_fail argument when ExternalTaskSensor runs in deferrable mode (#33196)
+- Fix handling of default value and serialization of Param class (#33141)
+- [Models] [Postgres] Check if the dynamically-added index is in the table schema before adding (#32731)
```
```suggestion
- Check if the dynamically-added index is in the table schema before adding (#32731)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,112 @@
+"""""""""""""
+- Use a trimmed version of README.md for PyPI (#33637)
+- Upgrade to ``Pydantic`` 2 (#33956)
+- Reorganize devel_only extra in airflow's setup.py (#33907)
```
```suggestion
- Reorganize ``devel_only`` extra in Airflow's setup.py (#33907)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,112 @@
+- Use a trimmed version of README.md for PyPI (#33637)
+- Upgrade to ``Pydantic`` 2 (#33956)
+- Reorganize devel_only extra in airflow's setup.py (#33907)
+- Suspend ``qubole`` provider (#33889)
```
I'm not sure this is really a core change even?

jedcunningham replied:
Meaning, we can probably skip this from the release notes for 2.7.1.

eladkal replied:
yeah should be removed from log

ephraimbuddy replied:
Looks like it added entry to setup.py

jedcunningham replied:
setup.py isn't that interesting imo, it just removes an item from `devel_only` which is main focused.

However, I guess you could make the argument it should be here because the `qubole` extra will go away.

eladkal replied:
Why would it go away? extra still exist

jedcunningham replied:
It would because the provider extras are added based on their presence in `provider_dependencies.json`, and it's being removed there.

Maybe when we suspend a provider it shouldn't be removed from there, idk 🤷‍♂️

ephraimbuddy replied:
> Maybe when we suspend a provider it shouldn't be removed from there, idk 

Breaking change :)

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,112 @@
+
+Doc only changes
+"""""""""""""""""
+- Explain the users how they can check if python code is top-level (#34006)
```
```suggestion
- Explain how users can check if python code is top-level (#34006)
```

potiuk committed:
Limit celery by excluding 5.3.2 and 5.3.3 (#34031)

There is a new database field introduced by Celery in 5.3.2 and
repeated in 5.3.3 wihch is not included in automated migrations,
so users upgrading celery might have failing celery installation.

The issue is already reported and acknowledged, so it is lilely
to be fixed in 5.3.4 - so excluding 5.3.2 and 5.3.4 is the best
approach.

(cherry picked from commit b6318ffabce8cc3fdb02c30842726476b7e1fcca)

dstandish committed:
Ensure that tasks wait for running indirect setup (#33903)

* move internal functions to methods -- no behavior change

* add setup constraint logic

* comments

* simplify

* simplify

* fix

* fix

* update tests

* static checks

* add constraint that setup tasks followed by ALL_SUCCESS rule

* add todo

* docs

* docs

* add test

* fix static check

(cherry picked from commit e75cecab72bd59ae9fa04631bdc7a27a745d61fe)
